Errors building Falco drivers for Linux kernels 5.10.x

**Describe the bug**

The driverkit internal scripts do not work for Linux kernels 5.10.x.


**How to reproduce it**

Given this configuration file (named "d510.yaml"):

```yaml
kernelversion: 1
kernelrelease: 5.10.0-1-amd64
target: debian
output:
  module: output/ae104eb20ff0198a5dcb0c91cc36c86e7c3f25c7/falco_debian_5.10.0-1-amd64_1.ko
  probe: output/ae104eb20ff0198a5dcb0c91cc36c86e7c3f25c7/falco_debian_5.10.0-1-amd64_1.o
```

Execute:

```console
make build
./_output/bin/driverkit docker --loglevel debug -c d510.yaml
```

**Expected behavior**

To build Falco drivers for the specified Linux Kernel 5.10.

**Screenshots**

```console
DEBU I+ make CC=/usr/bin/gcc-8 KERNELDIR=/usr/src/linux-headers-5.10.0-1-amd64 
DEBU Dmake -C /usr/src/linux-headers-5.10.0-1-amd64 M=/tmp/driver modules 
DEBU make[1]: Entering directory '/usr/src/linux-headers-5.10.0-1-amd64' 
DEBU make[1]: Leaving directory '/usr/src/linux-headers-5.10.0-1-amd64' 
DEBU Makefile:1: /usr/src/linux-headers-5.10.0-1-common/Makefile: No such file or directory 
DEBU make[1]: *** No rule to make target '/usr/src/linux-headers-5.10.0-1-common/Makefile'.  Stop. 
DEBU $make: *** [Makefile:7: all] Error 2
```

--- a/pkg/driverbuilder/builder/debian.go
+++ b/pkg/driverbuilder/builder/debian.go
@@ -162,16 +162,16 @@ func fetchDebianHeadersURLFromRelease(baseURL string, kr kernelrelease.KernelRel
 
 	// match for kernel versions like 4.19.0-6-amd64
 	extraVersionPartial := strings.TrimSuffix(kr.FullExtraversion, "-amd64")
-	matchExtraGroup := "amd64|common"
+	matchExtraGroup := "amd64"
+	matchExtraGroupCommon := "common"
 
 	// match for kernel versions like 4.19.0-6-cloud-amd64
 	if strings.Contains(kr.FullExtraversion, "-cloud") {
 		extraVersionPartial = strings.TrimSuffix(extraVersionPartial, "-cloud")
-		matchExtraGroup = "cloud-amd64|common"
+		matchExtraGroup = "cloud-amd64"
 	}
 
-	fullregex := fmt.Sprintf(rmatch, kr.Version, kr.PatchLevel, kr.Sublevel, extraVersionPartial, matchExtraGroup)
-	pattern := regexp.MustCompile(fullregex)
+	// download index
 	resp, err := http.Get(baseURL)
 	if err != nil {
 		return nil, err
@@ -181,15 +181,26 @@ func fetchDebianHeadersURLFromRelease(baseURL string, kr kernelrelease.KernelRel
 	if err != nil {
 		return nil, err
 	}
-	matches := pattern.FindAllStringSubmatch(string(body), 2)
+	bodyStr := string(body)
 
-	if len(matches) != 2 {
-		return nil, fmt.Errorf("kernel headers and kernel headers common not found")
+	// look for kernel headers
+	fullregex := fmt.Sprintf(rmatch, kr.Version, kr.PatchLevel, kr.Sublevel, extraVersionPartial, matchExtraGroup)
+	pattern := regexp.MustCompile(fullregex)
+	matches := pattern.FindStringSubmatch(bodyStr)
+	if len(matches) < 1 {
+		return nil, fmt.Errorf("kernel headers not found")
 	}
 
-	foundURLs := []string{fmt.Sprintf("%s%s", baseURL, matches[0][1])}
+	// look for kernel headers common
+	fullregexCommon := fmt.Sprintf(rmatch, kr.Version, kr.PatchLevel, kr.Sublevel, extraVersionPartial, matchExtraGroupCommon)
+	patternCommon := regexp.MustCompile(fullregexCommon)
+	matchesCommon := patternCommon.FindStringSubmatch(bodyStr)
+	if len(matchesCommon) < 1 {
+		return nil, fmt.Errorf("kernel headers common not found")
+	}
 
-	foundURLs = append(foundURLs, fmt.Sprintf("%s%s", baseURL, matches[1][1]))
+	foundURLs := []string{fmt.Sprintf("%s%s", baseURL, matches[1])}
+	foundURLs = append(foundURLs, fmt.Sprintf("%s%s", baseURL, matchesCommon[1]))
 
 	return foundURLs, nil
 }

